src/java/org/apache/cassandra/hints/HintsWriter.java

```diff
@@ -248,7 +248,10 @@ void append(Hint hint) throws IOException
                 updateChecksumInt(crc, hintSize);
                 out.writeInt((int) crc.getValue());
 
+                long startPosition = out.position();
                 Hint.serializer.serialize(hint, out, descriptor.messagingVersion());
+                long actualSize = out.position() - startPosition;
+                checkState(actualSize == hintSize);
```

Comment:
Not directly related to the patch, I guess, but seems harmless.

Reply:
Ah, yeah was debugging something I broke and felt it was worth leaving in.